wgpu-core/src/command/query.rs

```diff
@@ -404,38 +404,41 @@ impl Global {
             .check_usage(wgt::BufferUsages::QUERY_RESOLVE)
             .map_err(ResolveError::MissingBufferUsage)?;
 
-        let end_query = start_query + query_count;
-        if end_query > query_set.desc.count {
+        let end_query = u64::from(start_query)
+            .checked_add(u64::from(query_count))
+            .unwrap();
+        if end_query > u64::from(query_set.desc.count) {
             return Err(ResolveError::QueryOverrun {
                 start_query,
                 end_query,
                 query_set_size: query_set.desc.count,
             }
             .into());
         }
+        let end_query = u32::try_from(end_query).expect("`u32` overflow for `end_query`");
 
         let elements_per_query = match query_set.desc.ty {
             wgt::QueryType::Occlusion => 1,
             wgt::QueryType::PipelineStatistics(ps) => ps.bits().count_ones(),
             wgt::QueryType::Timestamp => 1,
         };
         let stride = elements_per_query * wgt::QUERY_SIZE;
-        let bytes_used = (stride * query_count) as BufferAddress;
+        let bytes_used: BufferAddress = u64::from(stride)
+            .checked_mul(u64::from(query_count))
+            .unwrap();
```

Comment:
The max value of stride is 64 * 8 so this should be fine too?

Reply:
I was on the fence on this one. Still, this shouldn't actually hurt anything in a `--release` build. Are you asking that this be changed?

Reply:
Discussed synchronously, an expect error message would help

Reply:
Added!